src/Npgsql/NpgsqlBatch.cs

```diff
@@ -52,6 +53,17 @@ public override int Timeout
             set => Transaction = (NpgsqlTransaction?)value;
         }
 
+        /// <summary>
+        /// Marks all of the batch's result columns as either known or unknown.
+        /// Unknown results column are requested them from PostgreSQL in text format, and Npgsql makes no
+        /// attempt to parse them. They will be accessible as strings only.
+        /// </summary>
+        public bool AllResultTypesAreUnknown
```

Comment:
I'm hesitating whether to make this internal for now (needed since this PR switches type-loading to use NpgsqlBatch, and that's needed there).

We can add this on individual NpgsqlBatchCommand instead, and so users can specify it on a command-per-command basis. But we can still have this as a batch-wide default - making internal defers all that for now.

Reply:
Made internal for now, files #3922 to track in the backlog.